js/bibox.js

```diff
@@ -1274,6 +1346,96 @@ module.exports = class bibox extends Exchange {
         };
     }
 
+    async transfer (code, amount, fromAccount, toAccount, params = {}) {
+        await this.loadMarkets ();
+        const currency = this.currency (code);
+        let method = undefined;
+        const request = {};
+        if (fromAccount === 'wallet' && toAccount === 'spot') {
+            method = 'privatePostAssetsTransferSpot';
+            request['type'] = 0;
+        } else if (fromAccount === 'spot' && toAccount === 'wallet') {
+            method = 'privatePostAssetsTransferSpot';
+            request['type'] = 1;
+        } else if (fromAccount === 'wallet' && toAccount === 'margin') {
+            method = 'privatePostCreditTransferAssetsBase2credit';
+        } else if (fromAccount === 'margin' && toAccount === 'wallet') {
+            method = 'privatePostCreditTransferAssetsCredit2base';
+        }
+        if (method === 'privatePostAssetsTransferSpot') {
+            request['symbol'] = currency['id'];
+            request['amount'] = this.currencyToPrecision (code, amount);
+        } else if (method === 'privatePostCreditTransferAssetsBase2credit' || method === 'privatePostCreditTransferAssetsCredit2base') {
+            request['coin_symbol'] = currency['id'];
+            request['amount'] = amount;
+            const symbol = this.safeString (params, 'symbol');
+            params = this.omit (params, 'symbol');
+            if (symbol === undefined) {
+                throw new ExchangeError ('to transfer to and from margin you must specify the `symbol` of the market for fixed margin');
+            }
+            const market = this.market (symbol);
+            request['pair'] = market['id'];
+        } else {
+            throw new ExchangeError ('this exchange only allows transfers between wallet and spot, and between wallet and margin');
+        }
+        const response = await this[method] (this.extend (request, params));
+        //
+        //    privatePostAssetsTransferSpot
+        //    {
+        //        "state":0,
+        //        "id":"794509221218521088"
+        //    }
+        //
+        //    privatePostCreditTransferAssetsBase2credit && privatePostCreditTransferAssetsCredit2Base
+        //    {
+        //        "result":"1370000000204",
+        //        "cmd":"transferAssets/base2credit",
+        //        "state":0
+        //    }
+        //
+        const transfer = this.parseTransfer (response, currency);
+        const transferOptions = this.safeValue (this.options, 'transfer');
+        const fillResponseFromRequest = this.safeValue (transferOptions, 'fillResponseFromRequest');
+        if (fillResponseFromRequest) {
```

Comment:
We need to test if it's explicitly undefined instead of testing for falsy values, like

```
        if (fillResponseFromRequest === undefined) {
```

-----------------

Also, is there any way to put this code into `parseTransfer`?

Reply:
I think instead of checking for undefined, I should default it to true, which is what I did on other PRs
```
const fillResponseFromRequest = this.safeValue (transferOptions, 'fillResponseFromRequest', true);
```

I agree it would be more elegant to have it inside `parseTransfer`, but without changing the function signature I don't see a way to pass in the values we need.